pkg/build/builder/source.go

```diff
@@ -229,6 +225,12 @@ func extractGitSource(gitClient GitClient, gitSource *api.GitBuildSource, revisi
 		if err := gitClient.SubmoduleUpdate(dir, true, true); err != nil {
 			return true, err
 		}
+
+	} else {
+		cloneOpts.Shallow = true
+		if err := gitClient.CloneWithOptions(dir, gitSource.URI, cloneOpts); err != nil {
+			return true, err
+		}
 	}
```

Comment:
I'm sorry, but does it work? Before we do clone, checkout and submodules update and now we have checkout, submodules update and clone at the end. Looks like we've changed the order of the operations and it may broke the program.